src/sentry/identity/pipeline.py

```diff
@@ -53,37 +58,42 @@ def get_provider(self, provider_key: str, **kwargs) -> PipelineProvider:
         return super().get_provider(provider_key)
 
     def finish_pipeline(self):
-        # NOTE: only reached in the case of linking a new identity
-        # via Social Auth pipelines
-        identity = self.provider.build_identity(self.state.data)
-
-        Identity.objects.link_identity(
-            user=self.request.user,
-            idp=self.provider_model,
-            external_id=identity["id"],
-            should_reattach=False,
-            defaults={
-                "scopes": identity.get("scopes", []),
-                "data": identity.get("data", {}),
-            },
-        )
-
-        messages.add_message(
-            self.request,
-            messages.SUCCESS,
-            IDENTITY_LINKED.format(identity_provider=self.provider.name),
-        )
-        metrics.incr(
-            "identity_provider_pipeline.finish_pipeline",
-            tags={
-                "provider": self.provider.key,
-            },
-            skip_internal=False,
-        )
-
-        self.state.clear()
-
-        # TODO(epurkhiser): When we have more identities and have built out an
-        # identity management page that supports these new identities (not
-        # social-auth ones), redirect to the identities page.
-        return HttpResponseRedirect(reverse("sentry-account-settings"))
+        with IntegrationPipelineViewEvent(
+            IntegrationPipelineViewType.IDENTITY_LINK,
+            IntegrationDomain.IDENTITY,
+            self.provider.key,
+        ).capture():
```

Comment:
i might pull this out because it's not entirely relevant to integrations in general, more like identity / auth?

Reply:
Is it only hit by integration installation like Azure when you sign in with them? or does it also get hit when you install the integration for ticket creation?

If the former, I think it makes sense to not do it here but if the latter we need it as part of the pipeline. Right?

Reply:
it should be hit as a part of the normal integration install pipeline via IdentityProviderPipeline as a NestedPipelineView